dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/interpretation/impl/DefaultInterpretationService.java

```diff
@@ -157,26 +213,151 @@ public void deleteInterpretation( Interpretation interpretation )
         return interpretationStore.getAllOrderedLastUpdated( first, max );
     }
 
+    @Override
+    public Set<User> getMentionedUsers( String text )
+    {
+        Set<User> users = new HashSet<>();
+        Matcher matcher = Pattern.compile( "(?:\\s|^)@([\\w+._-]+)" ).matcher( text );
+        while ( matcher.find() )
+        {
+            String username = matcher.group( 1 );
+            UserCredentials userCredentials = userService.getUserCredentialsByUsername( username );
+            if ( userCredentials != null )
+            {
+                users.add( userCredentials.getUserInfo() );
+            }
+        }
+        return users;
+    }
+
+    @Override
+    public void sendNotifications( Interpretation interpretation, InterpretationComment comment, Set<User> users )
+    {
+        if ( interpretation != null && users.size() > 0 )
```

Comment:
wouldn't it be better here to check for `intepretation == null || users.isEmpty()` and then return? now there is an entire indent level added because of this